src/Controllers/ResultController.php

```diff
@@ -51,8 +53,44 @@ public function getScorePage(RequestInterface $request)
             [
                 'session' => $this->session,
                 'user' => $userId,
+                'quizInstanceId' => $quizInstanceId,
                 'questions' => $questionInstanceEntities,
                 'answers' => $textInstnaces
             ]);
     }
+
+    public function scoreQuiz(RequestInterface $request)
+    {
+        $role = $this->session->get('role');
+
+        if ($role === null || $role === 'user') {
+            return $this->getErrorResponse();
+        }
+
+        $score = $request->getParameter('score');
+        $quizInstanceId = $request->getRequestParameters()['quizId'];
+        if (!$this->resultService->scoreResult($score, $quizInstanceId)) {
+            return $this->getErrorResponse();
+        }
+
+        return $this->getResultPage();
+    }
+
+    private function getErrorResponse(): Response
+    {
+        $response = new Response(Stream::createFromString(' '), []);
+        $response = $response->withStatus(301);
+        $response = $response->withHeader('Location', 'http://local.quiz.com/error/404');
+
+        return $response;
+    }
+
+    private function getResultPage(): Response
+    {
+        $response = new Response(Stream::createFromString(' '), []);
+        $response = $response->withStatus(301);
+        $response = $response->withHeader('Location', 'http://local.quiz.com/results/1');
+
+        return $response;
```

Comment:
a lot of copy/paste for redirects, move this to a method in abstract controller, or use a service/trait